### PR TITLE
Added Custom User Model

### PR DESCRIPTION
**Pull request by clintonb**
@rocha @dsjen

### CODE REVIEW

File: analytics_dashboard/analytics_dashboard/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        u'contenttypes.contenttype': {
+            'Meta': {'ordering': "('name',)", 'unique_together': "(('app_label', 'model'),)", 'object_name': 'ContentType', 'db_table': "'django_content_type'"},
+            'app_label': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
+            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
+            'model': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
+            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'})
+        }
+    }
+
+    complete_apps = ['analytics_dashboard']
```
Comment (dsjen) [inline analytics_dashboard/analytics_dashboard/migrations/0001_initial.py:98]: Github complains of a new line.

Comment (clintonb) [inline analytics_dashboard/analytics_dashboard/migrations/0001_initial.py:98]: Ignore it for migrations. They are generated and the new line doesn't really matter.

Comment (dsjen) [inline analytics_dashboard/analytics_dashboard/migrations/0001_initial.py:98]: Ah, ok.

**Comment by dsjen** [comment]
I don't know too much about this, but it looks good to me.

File: analytics_dashboard/analytics_dashboard/settings/base.py [hunk 2]
```diff
@@ -293,6 +293,8 @@
 DATE_FORMAT = 'F d, Y'
 
 ########## AUTHENTICATION
+AUTH_USER_MODEL = 'analytics_dashboard.User'
```
Comment (clintonb) [inline analytics_dashboard/analytics_dashboard/settings/base.py:296]: @dsjen This setting tells Django to use our custom user model.

File: analytics_dashboard/analytics_dashboard/models.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from django.conf import settings
+from django.contrib.auth.models import AbstractUser
+from django.db import models
+
+
+class User(AbstractUser):
```
Comment (clintonb) [inline analytics_dashboard/analytics_dashboard/models.py:6]: This is our custom user model, which simply adds a language field to the default fields (username, first_name, last_name, email).

File: analytics_dashboard/analytics_dashboard/migrations/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import SchemaMigration
+from django.db import models
+
+
+class Migration(SchemaMigration):
```
Comment (clintonb) [inline analytics_dashboard/analytics_dashboard/migrations/0001_initial.py:8]: Migration simply creates the table for the new user model. It's generated by South.

File: Makefile [hunk 5]
```diff
@@ -20,8 +20,8 @@ clean:
 
 test_python: clean
 	cd analytics_dashboard && ./manage.py test --settings=analytics_dashboard.settings.test --with-ignore-docstrings \
-		--exclude-dir=analytics_dashboard/settings --with-coverage --cover-inclusive --cover-branches \
-		--cover-html --cover-html-dir=$(COVERAGE)/html/ \
+		--exclude-dir=analytics_dashboard/settings --exclude-dir=analytics_dashboard/migrations --with-coverage \
```
Comment (clintonb) [inline Makefile:23]: Excluded the newly-created migrations directory from testing/coverage.

File: .pep8 [hunk 6]
```diff
@@ -1,4 +1,4 @@
 [pep8]
 ignore=E501
 max_line_length=119
-exclude=settings
+exclude=settings,migrations
```
Comment (clintonb) [inline .pep8:4]: Exclude migrations directories from PEP-8.

**Comment by clintonb** [comment]
@dsjen No worries. I added some comments to clarify. Let me know if you want to learn more.

**Comment by dsjen** [comment]
Thank you!  That was informative.

**Pull request merged**